deploy/docker-compose.yaml

```diff
@@ -0,0 +1,42 @@
+version: "3.7"
+
+volumes:
+  uploads:
+  data:
+
+services:
+  rocketchat:
+    image: registry.rocket.chat/rocketchat/rocket.chat:${RELEASE}
+    restart: on-failure
+    volumes:
+      - uploads:/app/uploads
+    environment:
+      MONGO_URL: mongodb://mongo:27017/rocketchat?replicaSet=rs0
+      MONGO_OPLOG_URL: mongodb://mongo:27017/local?replicaSet=rs0
```

Comment:
Is the link works if we add a password to the mongo?

Reply:
You'll have to change the uri to those information.

Reply:
need MONGODB_REPLICA_SET_KEY if set pwd

```
mongodb 09:46:53.85 
mongodb 09:46:53.85 Welcome to the Bitnami mongodb container
mongodb 09:46:53.85 Subscribe to project updates by watching https://github.com/bitnami/bitnami-docker-mongodb
mongodb 09:46:53.86 Submit issues and feature requests at https://github.com/bitnami/bitnami-docker-mongodb/issues
mongodb 09:46:53.86 
mongodb 09:46:53.86 INFO  ==> ** Starting MongoDB setup **
mongodb 09:46:53.89 INFO  ==> Validating settings in MONGODB_* env vars...
mongodb 09:46:53.89 ERROR ==> In order to configure MongoDB replica set authentication you need to provide the MONGODB_REPLICA_SET_KEY on every node, specify MONGODB_ROOT_PASSWORD in the primary node and MONGODB_INITIAL_PRIMARY_ROOT_PASSWORD in the rest of nodes
```

Reply:
Could you give me an example of setting with a mongo pwd. 
I tried but failed. Thanks.

```
MongoParseError: Unescaped slash in userinfo section
    at parseConnectionString (/app/bundle/programs/server/npm/node_modules/meteor/npm-mongo/node_modules/mongodb/lib/core/uri_parser.js:611:21)
    at connect (/app/bundle/programs/server/npm/node_modules/meteor/npm-mongo/node_modules/mongodb/lib/operations/connect.js:281:3)
    at /app/bundle/programs/server/npm/node_modules/meteor/npm-mongo/node_modules/mongodb/lib/mongo_client.js:256:5
    at maybePromise (/app/bundle/programs/server/npm/node_modules/meteor/npm-mongo/node_modules/mongodb/lib/utils.js:685:3)
    at MongoClient.connect (/app/bundle/programs/server/npm/node_modules/meteor/npm-mongo/node_modules/mongodb/lib/mongo_client.js:252:10)
    at Function.MongoClient.connect (/app/bundle/programs/server/npm/node_modules/meteor/npm-mongo/node_modules/mongodb/lib/mongo_client.js:476:22)
    at new MongoConnection (packages/mongo/mongo_driver.js:208:11)
    at new MongoInternals.RemoteCollectionDriver (packages/mongo/remote_collection_driver.js:4:16)
    at Object.<anonymous> (packages/mongo/remote_collection_driver.js:38:10)
    at Object.defaultRemoteCollectionDriver (packages/underscore.js:784:19)
    at new Collection (packages/mongo/collection.js:97:40)
    at new AccountsCommon (packages/accounts-base/accounts_common.js:23:18)
    at new AccountsServer (packages/accounts-base/accounts_server.js:24:5)
    at packages/accounts-base/server_main.js:7:12
    at module (packages/accounts-base/server_main.js:19:1)
    at fileEvaluate (packages/modules-runtime.js:336:7)
/app/bundle/programs/server/node_modules/fibers/future.js:280
						throw(ex);
```

Reply:
I tried this config and it works.

.env
```
### Rocket.Chat configuration

# Rocket.Chat version
# see:- https://github.com/RocketChat/Rocket.Chat/releases
RELEASE=4.1.2
#RELEASE=latest
# MongoDB endpoint (include ?replicaSet= parameter)
#MONGO_URL=mongodb://rocketchat:xxx@mongo:27017/rocketchat?replicaSet=rs0
# MongoDB endpoint to admin database
#MONGO_OPLOG_URL=mongodb://rocketchat:xxx@mongo:27017/admin
# URL used to access your Rocket.Chat instance
ROOT_URL=https://rocketchat.fjlandao.com
# Port Rocket.Chat runs on (in-container)
PORT=3000
# Port on the host to bind to
HOST_PORT=3000

### MongoDB configuration
# See:- https://hub.docker.com/r/bitnami/mongodb
ALLOW_EMPTY_PASSWORD=no
MONGODB_ROOT_USER=root
MONGODB_ROOT_PASSWORD=123456
MONGODB_USERNAME=rocketchat
MONGODB_PASSWORD=123456
MONGODB_NAME=rocketchat
MONGODB_REPLICA_KEY=replicasetkey123
```

docker-compose.yaml
```
version: "3.7"

volumes:
  uploads:
  data:

services:
  rocketchat:
    image: registry.rocket.chat/rocketchat/rocket.chat:${RELEASE}
    restart: on-failure
    volumes:
      - ./uploads:/app/uploads
    environment:
      MONGO_URL: mongodb://${MONGODB_ROOT_USER}:${MONGODB_ROOT_PASSWORD}@mongo:27017/rocketchat?replicaSet=rs0&authSource=admin
      MONGO_OPLOG_URL: mongodb://${MONGODB_ROOT_USER}:${MONGODB_ROOT_PASSWORD}@mongo:27017/local?replicaSet=rs0&authSource=admin
      ROOT_URL: ${ROOT_URL}
      PORT: ${PORT}
    depends_on: [mongo]
    ports:
      - 3000:3000
    labels:
      - "traefik.backend=rocketchat"
      - "traefik.frontend.rule=Host: ${ROOT_URL}"

  mongo:
    image: bitnami/mongodb:4.4.10
    restart: on-failure
    volumes:
      # - ./data/db:/data/db
      - ./data:/bitnami/mongodb/data
    environment:
      MONGODB_REPLICA_SET_MODE: primary
      MONGODB_REPLICA_SET_NAME: rs0
      MONGODB_PORT_NUMBER: "27017"
      MONGODB_INITIAL_PRIMARY_HOST: mongo
      MONGODB_INITIAL_PRIMARY_PORT_NUMBER: "27017"
      MONGODB_ADVERTISED_HOSTNAME: mongo
      MONGODB_ENABLE_JOURNAL: "true"
      ALLOW_EMPTY_PASSWORD: ${ALLOW_EMPTY_PASSWORD}
      MONGODB_ROOT_USER: ${MONGODB_ROOT_USER}
      MONGODB_ROOT_PASSWORD: ${MONGODB_ROOT_PASSWORD}
      MONGODB_USERNAME: ${MONGODB_USERNAME}
      MONGODB_PASSWORD: ${MONGODB_PASSWORD}
      MONGODB_DATABASE: ${MONGODB_NAME}
      MONGODB_REPLICA_SET_KEY: ${MONGODB_REPLICA_KEY}
      MONGODB_EXTRA_FLAGS: --wiredTigerCacheSizeGB=2
      MONGODB_EXTRA_FLAGS: --storageEngine=wiredTiger
      MONGODB_EXTRA_FLAGS: --oplogSize=128
    # command: mongod --oplogSize 128 --replSet rs0 --storageEngine=wiredTiger
    ports:
      - 27017:27017
    labels:
      - "traefik.enable=false"
```